tqdm/_tqdm.py

```diff
@@ -734,6 +752,10 @@ def __iter__(self):
                                 else smoothing * delta_t / delta_it + \
                                 (1 - smoothing) * avg_time
 
+                        # Acquire locks if parallel bars
```

Comment:
We tried everything we could imagine, including what you propose, and it lowers drastically the performance of ``__iter__()``.

I think there is no way around, ``__iter__()`` is super efficient because once it's launched it works in its own virtual routine with minimal calls to external functions (see how most ``self`` variables are stored in method's local variables before the for loop), so any attempt to unify ``__iter__`` and ``update`` is bound to fail IMO (except maybe if you're a Python wizard and you dynamically rewrite the method on first call depending on whether it's __iter__ or not, but then I think it would be way more complex to maintain than simple code duplication).